packages/dds/map/src/directory.ts

```diff
@@ -1636,7 +1636,12 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 		if (!this.needProcessSubDirectoryOperation(msg, op, local, localOpMetadata)) {
 			return;
 		}
-		this.createSubDirectoryCore(op.subdirName, local, msg.sequenceNumber, msg.clientId);
+		this.createSubDirectoryCore(
+			op.subdirName,
+			local,
+			msg.sequenceNumber,
+			msg.clientId ?? "server",
```

Comment:
If we ever get a null client ID here would be pretty wild.  I think this should be guarded earlier in the pipe (maybe `SharedObject.process()` or even earlier).  If you don't want to mess with that right now though, then probably prefer to assert here rather than filling in "server".  I think similar for the other instances below.

Reply:
maybe. i don't know where we'll go with external data sync stuff. potentially i could see server generate ops for append only operations.

Reply:
I think if/when we add functionality that relies on server ops making it through to here would be the time to add handling like this - I think we should assert until the behavior is well-defined (as the detached behavior is (I hope) well-defined)

Reply:
i don't want to add an assert. i'm pretty confident it never happens, but want to be cautions. for now i'll just make the case explicit, rather than implicit as is pretty much is now.